src/components/metadataSelects/CategoryComboSelect.tsx

```diff
@@ -0,0 +1,132 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { SearchableSingleSelect } from '../SearchableSingleSelect'
+
+interface CategoryComboQueryResult {
+    categoryCombos: {
+        pager: {
+            pageCount: number,
+            pageSize: number,
+            page: number,
+            total: number,
+        },
+        categoryCombos: Array<{
+            id: string,
+            displayName: string,
+        }>,
+    }
+}
```

Comment:
We have types for this. Currently the generic-result types are only for gists. And I think this also should use gist because it's faster - regular API should just be used if you need very nested responses.

I'm not super happy yet with `useModelGist`, and I think we will migrate to use react-query directly regardless. But it should be easier to use for now.

```ts
import { CategoryCombo, GistCollectionResponse } from '../../types/models'

// reuse filterFields for model and fields in query
const filterFields = ['id', 'name'] as const //(name is translated by default in /gist)
type FilteredCategoryCombo = Pick<CategoryCombo, typeof filterFields[number]>

type CategoryComboQueryResult = GistCollectionResponse<FilteredCategoryCombo>
```

Reply:
When I use the proposed types, I'm getting type errors:

![image](https://github.com/dhis2/maintenance-app-beta/assets/10101831/c2a79739-8edd-443e-b988-2942c9cdf2ef)